crates/core/src/transport/symmetric_message.rs

```diff
@@ -79,18 +89,14 @@ impl SymmetricMessage {
             )),
         },
     };
-
-    const ACK_OK: SymmetricMessage = SymmetricMessage {
-        message_id: FIRST_MESSAGE_ID,
-        confirm_receipt: None,
-        payload: SymmetricMessagePayload::AckConnection { result: Ok(()) },
-    };
 }
 
 #[derive(Serialize, Deserialize)]
 pub(super) enum SymmetricMessagePayload {
     AckConnection {
-        result: Result<(), Cow<'static, str>>,
+        // if we successfully connected to a remote we attempt to connect to initially
+        // then we return our TransportPublicKey so they can enroute other peers to us
+        result: Result<TransportPublicKey, Cow<'static, str>>,
```

Comment:
Isn't the joiner's `TransportPublicKey` sent in the `ConnectionRequest::StartReq.target`? I'm not sure why it needs to be sent as part of the transport layer.